### PR TITLE
[PRD-6041] Data Format specified in the Parameter is not working as e…

### PR DESCRIPTION
**Pull request by LeonardoCoelho71950**
…xpected.

Merge with: https://github.com/pentaho/pentaho-platform-plugin-common-ui/pull/1424

@pentaho-lmartins @dcleao

### CODE REVIEW

File: core-js/src/main/javascript/cdf/components/TextInputComponent.js [hunk 1]
```diff
@@ -24,9 +24,20 @@ define([
       myself._addHtmlToPlaceholder();
 
       var el = $("#" + myself.name);
-      el.val(myself.dashboard.getParameterValue(myself.parameter));
+      if (myself.formatter) {
```
Comment (dcleao) [inline core-js/src/main/javascript/cdf/components/TextInputComponent.js]: Please read the parameter value a single time and declare the `val` variable. For example:
```js
var value = myself.dashboard.getParameterValue(myself.parameter);
if (myself.formatter) {
  value = myself.formatter.format(value);
}
el.val(value);
```
```

File: core-js/src/main/javascript/cdf/components/TextInputComponent.js [hunk 2]
```diff
@@ -24,9 +24,20 @@ define([
...
+        // Format user input
+        val = myself.dashboard.getParameterValue(myself.parameter);
+        el.val(myself.formatter.format(val));
+      } else {
+        el.val(myself.dashboard.getParameterValue(myself.parameter));
+      }
+
       el.change(function() {
         if(myself.dashboard.getParameterValue(myself.parameter) !== el.val()) {
+          if (myself.formatter && myself.formatter.format(el.val())) {
```
Comment (dcleao) [inline core-js/src/main/javascript/cdf/components/TextInputComponent.js]: Formatting an already formatted value mixed with user input?

File: core-js/src/main/javascript/cdf/components/TextInputComponent.js [hunk 3]
```diff
@@ -39,7 +50,12 @@ define([
     },
 
     getValue: function() {
-      return $("#" + this.name).val();
+      val = $("#" + this.name).val();
```
Comment (dcleao) [inline core-js/src/main/javascript/cdf/components/TextInputComponent.js]: Global variable.

File: core-js/src/main/javascript/cdf/components/TextInputComponent.js (hunk 1 above)
Comment (LeonardoCoelho71950) [reply core-js/src/main/javascript/cdf/components/TextInputComponent.js]: Done

File: core-js/src/main/javascript/cdf/components/TextInputComponent.js (hunk 3 above)
Comment (LeonardoCoelho71950) [reply core-js/src/main/javascript/cdf/components/TextInputComponent.js]: Done

**Pull request merged**